### PR TITLE
176895077 google calendar link should be visible only with registration

### PR DESCRIPTION
**Pull request by rajavi-mishra**
### Proposed Changes

1. For a scheduled event, the event page displays the "Add to Google Calendar (beta)" button only if the user is signed in
2. Tests have been added for the same

Contributor
@rajavi-mishra Rajavi Mishra

### High Fidelity Screenshots

**Screenshot 1**
Description: For a scheduled event, when the user is signed in, the user can see the "Add to Google Calendar (beta)" button 
Screen Shot 2021-04-25 at 1 06 43 AM

**Screenshot 2**
Description: For a scheduled event, when the user is not signed in, the user cannot see the "Add to Google Calendar (beta)" button
Screen Shot 2021-04-25 at 1 07 08 AM

### CODE REVIEW

File: spec/features/proposals_spec.rb [hunk 1]
```diff
@@ -212,4 +212,26 @@
...
+    before(:each) do
+      @user = create(:user)
+      @program = conference.program
+      @selected_schedule = create(:schedule, program: @program)
+      @program.update_attributes!(selected_schedule: @selected_schedule)
+      @scheduled_event1 = create(:event, program: @program, state: 'confirmed', abstract: '`markdown`')
+      @event_schedule1 = create(:event_schedule, event: @scheduled_event1, schedule: @selected_schedule, start_time: conference.start_hour + 1.hour)
+    end
+
+    scenario 'for a scheduled event, can add an event to google calendar if signed in', feature: true, js: true do
```
Comment (CactusPuppy) [inline spec/features/proposals_spec.rb]: The `js: true` is unnecessary, as this test does not rely on JavaScript. Unnecessarily starting the JavaScript framework to run a test slows down the test suite overall.

File: spec/features/proposals_spec.rb [hunk 2]
```diff
@@ -212,4 +212,26 @@
...
+      sign_in @user
+      visit conference_program_proposal_path(conference.short_title, @scheduled_event1.id)
+      expect(page).to have_content('Add to Google Calendar (beta)')
+    end
+
+    scenario 'for a scheduled event, cannot add an event to google calendar if not signed on', feature: true, js: true do
```
Comment (CactusPuppy) [inline spec/features/proposals_spec.rb]: See the above comment on `js: true`.

**Review by CactusPuppy — CHANGES_REQUESTED** [review]
Small thing: we shouldn't initialize the JavaScript framework for tests which don't need it, as that startup costs time.

File: spec/features/proposals_spec.rb [hunk 3]
```diff
@@ -212,4 +212,26 @@
       expect(page.find('#event_submission_text').value).to eq(event_type.submission_instructions)
     end
   end
+
+  context 'as a user, looking at a conference with scheduled events' do
+    before(:each) do
+      @user = create(:user)
```
Comment (kingdish) [inline spec/features/proposals_spec.rb]: I think there's already a `participant` we can use on line 10.

**Review by kingdish — CHANGES_REQUESTED** [review]

**Pull request merged**